### PR TITLE
Add OpenAI pipeline examples with LangChain

### PR DESCRIPTION
**Pull request by maximearmstrong**
## Summary & Motivation

This PR adds examples to support the new Dagster + OpenAI blog post

## How I Tested These Changes

BK

### CODE REVIEW

**Comment by maximearmstrong** [comment]
* **#20279** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/20279?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @maximearmstrong and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: examples/with_openai/with_metadata.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+from dagster_openai import OpenAIResource
+from langchain.chains.qa_with_sources import load_qa_with_sources_chain
+from langchain.docstore.document import Document
+from langchain.embeddings.openai import OpenAIEmbeddings
+from langchain.text_splitter import CharacterTextSplitter
+from langchain.vectorstores.faiss import FAISS
+from langchain_community.llms import OpenAI
+
+
+@asset
```
Comment (yuhan) [inline examples/with_openai/with_metadata.py:23]: same, nit: could be `compute_kind="GitHub"`

File: examples/with_openai/with_partitions.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+from langchain.text_splitter import CharacterTextSplitter
+from langchain.vectorstores.faiss import FAISS
+from langchain_community.llms import OpenAI
+
+docs_partitions_def = StaticPartitionsDefinition(["guides", "integrations"])
+
+SEARCH_INDEX_FILE = "search_index.pickle"
+
+
+@asset(partitions_def=docs_partitions_def)
```
Comment (yuhan) [inline examples/with_openai/with_partitions.py:31]: same nit as before.

**Review by yuhan — COMMENTED** [review]
code looks good.

in terms of the example structure, i was thinking, instead of showing individual single files for each sections, should we just simply make it as a full project via the `dagster project scaffold` cli? so that:
- the project structure would be consistent with other ones in the examples dir
- we don't need to deal with that separate `requirements.txt` file.
- also, maybe we can also get rid of the multi with_x.py files - we could just keep the full project with the final `with_configs.py` (the one with sensors) and other scaffolding files, with the `baseline.py` as a separate standalone file in the project root. -- so, for users who want to follow along, they can build from `baseline.py`; but for users who want to run it quickly, they can just `dagster dev` to get the final result running quickly.

**Review by yuhan — COMMENTED** [review]
this is more for simplicity as this project isn't very complex, so could we consolidate some files:
- merge `sensors` to `__init__.py`
- merge `jobs.py` to `assets.py` or `__init__.py` - i think it can be either and it'd depend on which is easier for imports/reference.
- merge "partitions.py" to `assets.py`

i like the util.py separation as that makes it clear whats an asser vs whats a helper func.

**Review by yuhan — APPROVED** [review]

**Pull request merged**